Client/Application/Delegates/SceneDelegate.swift

```diff
@@ -55,39 +78,436 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+                }
+                
+                return
+            default:
+                break
+        }
+        
+        func switchToTabForIntentURL(intentURL: String?) {
+            guard let siteURL = intentURL, let url = URL(string: siteURL) else {
+                return
+            }
+
+            if let browserViewController = scene.browserViewController {
+                browserViewController.switchToTabForURLOrOpen(
+                    url,
+                    isPrivate: Preferences.Privacy.privateBrowsingOnly.value,
+                    isPrivileged: false)
+            }
+            return
+        }
+        
+        if let intent = userActivity.interaction?.intent as? OpenWebsiteIntent {
+            return switchToTabForIntentURL(intentURL: intent.websiteURL)
+        }
+        
+        if let intent = userActivity.interaction?.intent as? OpenHistoryWebsiteIntent {
+            return switchToTabForIntentURL(intentURL: intent.websiteURL)
+        }
+        
+        if let intent = userActivity.interaction?.intent as? OpenBookmarkWebsiteIntent {
+            return switchToTabForIntentURL(intentURL: intent.websiteURL)
+        }
+        
+        return
+    }
+    
+    func windowScene(_ windowScene: UIWindowScene, performActionFor shortcutItem: UIApplicationShortcutItem, completionHandler: @escaping (Bool) -> Void) {
+        
+        if let browserViewController = windowScene.browserViewController {
+            QuickActions.sharedInstance.handleShortCutItem(shortcutItem, withBrowserViewController: browserViewController)
+            completionHandler(true)
+        } else {
+            completionHandler(false)
+        }
+    }
+}
+
+// MARK: - Root View Controller Animations
+extension SceneDelegate: UINavigationControllerDelegate {
+    func navigationController(_ navigationController: UINavigationController, animationControllerFor operation: UINavigationController.Operation, from fromVC: UIViewController, to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        switch operation {
+        case .push:
+            return BrowserToTrayAnimator()
+        case .pop:
+            return TrayToBrowserAnimator()
+        default:
+            return nil
+        }
+    }
+}
+
+extension SceneDelegate {
+    static let profile: Profile = {
+        BrowserProfile(localName: "profile")
+    }()
+    
+    static let diskImageStore: DiskImageStore? = {
+        do {
+            return try DiskImageStore(files: profile.files,
+                                      namespace: "TabManagerScreenshots",
+                                      quality: UIConstants.screenshotQuality)
+        } catch {
+            log.error("Failed to create an image store for files: \(profile.files) and namespace: \"TabManagerScreenshots\": \(error.localizedDescription)")
+            assertionFailure()
+        }
+        return nil
+    }()
```

Comment:
Why are these static, couldn't they exist in AppDelegate and pass into each scene?

Reply:
Not sure how to pass the data from `AppDelegate` to each scene.